<launch>

    <arg name="robot_name" default="sam"/>

    <node name="pid_actuator_vbs_alt" pkg="sam_basic_controllers" type="pid_actuator_trim" ns="vbs_alt" >
      <param name="topic_from_controller" value="control_action" />
      <param name="topic_to_actuator" value="/actuator_input" />
      <param name="pid_enable_topic" value="/$(arg robot_name)/ctrl/vbs_alt/pid_enable" />
      <param name="abort_topic" value="/$(arg robot_name)/abort" />
      <param name="limit_between_setpoints" value="1" />
      <param name="loop_freq" value="50" />
      <remap from="/actuator_input" to="/$(arg robot_name)/core/vbs_cmd"/>
    </node>

    <node name="vbs_alt_pid" pkg="pid" type="controller" ns="vbs_alt" output="screen" >
      <param name="Kp" value="-50.0" />
      <param name="Ki" value="-0.2" />
      <param name="Kd" value="-5" />
      <param name="upper_limit" value="50.0" />
      <param name="lower_limit" value="-50.0" />
      <param name="windup_limit" value="100" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
      <!--ADD ALTITUDE FEEDBACK TOPIC!! /-->
      <param name="topic_from_plant" value="/$(arg robot_name)/dr/altitude" />
      <!--param name="topic_from_plant" value="/$(arg robot_name)/ctrl/odom_listener/alt_feedback" /-->
      <param name="topic_from_controller" value="control_action" />
      <remap from="setpoint" to="setpoint" />
      <remap from="pid_enable" to="pid_enable" />
     </node>

</launch>
